src/west/app/main.py

```diff
@@ -8,7 +8,7 @@
 
 '''Zephyr RTOS meta-tool (west) main module
 
-Nothing in here is public API.
+Only the WestApp class and the run method in here is public API.
```

Comment:
> To me, an object oriented API implies that I can instantiate multiple WestApp objects and use them at the same time.
>
> This is not true, because of #149.

That's a valid point.


> > Secondly, main() itself may be ambiguous.
> If that's a problem, import ... as is a Pythonic fix:

I know you can `import ... as ...`, and that is exactly useful when there are colliding names (or you want to import an alternative implementation of existing API, but the existing code is already referring the name of previous module implementing the API).

But `main` is such common name, that I found it more correct to use a `West` specific API, and not going through a `import ... as ...` cause with `main` it makes the code look as we are doing something we shouldn't be doing imho.
But as long as #149 is open, using `main()` is fine.

We could also choose to fix #149.
If #149 is fixed, which version would you prefer, `main()` or `WestApp().run()` ?

I would like to avoid switching from `main` --> `WestApp().run()` later, when #149 is fixed.
But if you prefer `main()` in any case, then I will switch back.

Reply:
I don't think we have time to fix 149 any time soon unfortunately, so would you mind switching back to main()? We can always document WestApp later and I think there won't be any harm done if we switch; we just need to maintain both. Although honestly I think of WestApp as much more of an implementation detail than the existence of a main() function for a program with a CLI.

Reply:
done. switched back to `main()`

Reply:
Thanks a lot for bearing with me on this.

If you can't get the API docs working with shippable on the zephyr PR, it's fine; I will remember this when I'm writing up the 0.8.0 docs.

Reply:
No problem. I prefer it takes time to settle and then do code changes, compared to having to change to often.

I think it was good discussion, although I still prefer the `WestApp` version, but I do acknowledge we should get #149 fixed.

Reply:
btw. the docs build correctly locally:
![image](https://user-images.githubusercontent.com/9802583/84985886-ddc4b900-b13d-11ea-93ee-e34ee9cd9abd.png)

Reply:
> btw. the docs build correctly locally:

Sure, we just may not be able to get this built in the zephyr CI system and nightlies until the next west release.